iothub/device/src/MethodInvokeRequest.cs

```diff
@@ -23,16 +23,19 @@ internal class MethodInvokeRequest
         /// <exception cref="ArgumentException">If <b>methodName</b> is null or whitespace</exception>
         public MethodInvokeRequest(string methodName, string payload, TimeSpan? responseTimeout, TimeSpan? connectionTimeout)
         {
-            ValidatePayloadIsJson(payload);
+            if (!string.IsNullOrEmpty(payload))
+            {
+                ValidatePayloadIsJson(payload);
+                this.Payload = new JRaw(payload);
+            }
             if (string.IsNullOrWhiteSpace(methodName))
```

Comment:
Two observations:
1. Unless there's any reason against it, please move the `methodName` parameter check above. We should validate params from left to right.
2. The message in line 33 is wrong - please replace with ArgumentNullException(nameof(methodName)); (Nit: why perform such validation in the first place? The method is on an internal class - our application should use internal APIs correctly.)

Reply:
1. Done.
2. Replaced. From the perspective of building secure layers, I think checking in the internal class makes sense.